### ISSUE TITLE
Continuation access token use outside of continuation requests

### ISSUE THREAD

**Issue by jricher** [ietf-wg-gnap/gnap-core-protocol#66]
§3.1 Request Continuation: Editor's note:

> This access token MUST NOT be usable at resources outside of the AS.

Is this a restriction we want to enforce?

**Comment by jricher** [comment]
Editors suggest moving this text to a security consideration as defining this normatively would define internal behavior to the AS that does not affect interoperability.

**Comment by yaronf** [comment]
Disagree. we should define security-critical functionality in the **body** of the protocol, even if it does not affect interoperability. And even if it cannot be enforced.

**Comment by fimbault** [comment]
Will at least be handled within a dedicated security consideration
@yaronf If we find that we can put normative requirements into the body, we'll do so. Right now editors feel this specific issue can be closed as it has been referenced in #135

**Comment by yaronf** [comment]
No, please don't shovel individual security issues into a "security" uber-issue. #135 is about formal analysis of the protocol, that's a whole different matter.

**Comment by fimbault** [comment]
Then i would suggest that we put this issue in "need text", but it feels strange to describe functionality that can't be enforced, does it not?

**Comment by jricher** [comment]
I believe @yaronf's position is to leave the normative requirement as it stands in the text right now, without text changes (which would mean closing this issue without a PR). The editors' original proposed solution to this issue was to add security considerations about the token but remove the normative requirement because of its lack of enforceability, which would require a PR. The editors' new comment above is that we will at least add security considerations for this item regardless of what happens with the normative statement.

**Comment by yaronf** [comment]
Yes, this is my position. Security-critical statements should be normative, and should be included in the main text where it makes best sense for an implementer. Regardless of enforceability. 

As a reminder, many cryptographic checks (the recipient MUST validate this hash/signature/format) are hard or impossible to enforce.

**Comment by fimbault** [comment]
"As a reminder, many cryptographic checks (the recipient MUST validate this hash/signature/format) are hard or impossible to enforce."
This is very different as this refers to what the client should/must do. So there's really an entity that is concerned with the requirement. 

While in the specific case related to the issue, I really have no idea who's targeted by the requirement, which seems nothing more than a trust model.

**Comment by yaronf** [comment]
@fimbault I don't understand your point. The AS defines different tokens so that they are trusted for different resources. It can implement this trust different ways: e.g. by marking the tokens differently in its internal database, or by signing different attributes within the JWT that's the token. This requirement is a constraint on how the AS should manage this authorization. I don't think it's a requirement on the Client at all.

**Comment by fimbault** [comment]
@yaronf Maybe I didn't get what you meant by the cryptographic check you're talking about. But it seems we're diverging a bit from the original issue. 

The question is whether we can close this issue as suggested by the editorial team, knowing that we keep in mind your point about making sure security requirements are well defined in the core of the spec.

**Comment by yaronf** [comment]
It's not clear what the team is proposing now. If the proposal is (1) to make this constraint non-normative and (2) to move it outside the main text, I disagree with (1) strongly and with (2) less strongly...

**Comment by fimbault** [comment]
(1) MUST here doesn't seem directly enforceable. Or we might need to add text to explain what needs to be done about it.
(2) In practice, the proposal would be that it is still be in the main text, but in a dedicated section related to the trust model.

**Comment by aaronpk** [comment]
See also #149 

Let's rephrase this to the inverse so that it is something that is actually testable.

Access tokens other than the continuation access token (name tbd based on #149) MUST NOT be usable for continuation requests. 

In other words, regular access tokens used for accessing resources MUST NOT be usable for continuation requests.
